### PR TITLE
Add task registry

### LINKED ISSUE #32: Add ability to switch task at runtime
This includes switching the simulation as the new task should run in a fresh Simualtion

### PR DESCRIPTION
**Pull request by buehlefs**
Add a task registry and handle task registration and task switches correctly.

Closes #32

### CODE REVIEW

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationProxy.java [hunk 1]
```diff
@@ -145,14 +191,60 @@ public void simulationSpeedChange(final int value) {
     
     @Override
     public void setTaskSelectorListener(TaskSelectorListener listener) {
-        if ((this.taskSelectorListener == null) || (listener == null)) {
-            this.taskSelectorListener = listener;
-        } else throw new ListenerSetException();
+        this.taskRegistry.setTaskSelectorListener(listener);
```
Comment (mee4895) [inline ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationProxy.java]: I want to bitch about this, but I fear I can't ...

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/tasks/StandardTaskRegistry.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     * @param listener
+     *     the listener to set; use null to remove listener
+     * @throws ListenerSetException
+     *     if the listener is already set and the provided listener is not {@code null}.
+     */
+    public void setTaskSelectorListener(TaskSelectorListener listener) {
+        if ((this.changeListener == null) || (listener == null)) {
+            this.changeListener = listener;
+        } else throw new ListenerSetException();
+    }
```
Comment (mee4895) [inline ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/tasks/StandardTaskRegistry.java]: As I thought there is no need to blindly bitch about stuff, but out of curiosity why that change?

File: ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/GameWindowFactory.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public static GameWindow createGameWindow(final SimulationProxy simulationProxy)
         }
         
         final SwingTextureRegistry textureRegistry = new SwingTextureRegistry();
-        final SwingPlayfieldDrawer playfieldDrawer = new SwingPlayfieldDrawer();
+        final SwingPlayfieldDrawer playfieldDrawer = new SwingPlayfieldDrawer(simulationProxy, textureRegistry);
```
Comment (mee4895) [inline ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/GameWindowFactory.java]: Nice to see that change, down with `.initialize(this);`

File: ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/internal/SwingGameWindow.java [hunk 4]
```diff
@@ -106,7 +106,7 @@ public void setWindowTitle(final String title) {
     public void start() {
         // init jFrame
         this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
-        this.playfieldDrawer.initialize(this);
+        this.playfieldDrawer.initialize();
```
Comment (mee4895) [inline ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/internal/SwingGameWindow.java]: At least a step in the right direction :stuck_out_tongue_winking_eye:

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/tasks/StandardTaskRegistry.java (hunk 2 above)
Comment (buehlefs) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/tasks/StandardTaskRegistry.java]: Only the task registry knows when new tasks are registered. The change event has to be triggered in this class.

**Pull request merged**